sourcegraph/cody
# add context logging for autoedits

## Context
This PR adds the logging for various context sources for OSS repos, dotComUsers. In addition, we define a feature flag, and only if the user is in the variant, we collect the data, this way with the help of feature flag we can control the percent of logging without changing the code.
The data is particular useful for evaluation and finetuning of autoedits, and context source evaluation for completions. 

## Test plan
Manual testing:
1. Override in the feature flag and observe the change in the data collection
2. Added CI tests
